plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -58,8 +61,15 @@ sealed class AbiClassInfo {
  */
 class JvmAbiClassBuilderInterceptor(
     private val removeDataClassCopyIfConstructorIsPrivate: Boolean,
+    private val removePrivateClasses: Boolean,
 ) : ClassGeneratorExtension {
-    val abiClassInfo: MutableMap<String, AbiClassInfo> = mutableMapOf()
+    private var abiClassInfoBuilder = AbiClassInfoBuilder()
+
+    fun buildAbiClassInfoAndReleaseResources(): Map<String, AbiClassInfo> {
+        return abiClassInfoBuilder.buildClassInfo().also {
+            abiClassInfoBuilder = AbiClassInfoBuilder()
+        }
+    }
```

Comment:
I'm storing more info about classes to be able to keep private parents of annotations and exposed private interfaces (their parents too, if any). Additionally I get better encapsulation == easier to read/follow the code. Please make sure it is ok approach.